### PR TITLE
DOC-826 Add support Cloud Spanner driver

### PR DESCRIPTION
**Pull request by asimms41**
## Description

Resolves [DOC-826](https://redpandadata.atlassian.net/browse/DOC-826)
Review deadline: 11th December

Adds Google Cloud spanner driver details to all of the following pages:

## Page previews

- [`sql` cache](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/caches/sql/)

Inputs
- [`sql_raw`](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/inputs/sql_raw/)
- [`sql_select`](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/inputs/sql_select/)

Outputs
- [`sql`](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/outputs/sql/)
- [`sql_insert`](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/outputs/sql_insert/)
- [`sql_raw`](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/outputs/sql_raw/)

Processors
- [`sql`](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/processors/sql/)
- [`sql_insert`](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/processors/sql_insert/)
- [`sql_raw`](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/processors/sql_raw/)
- [`sql_select`](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/processors/sql_select/)

## Checks

- [ ] New feature
- [x] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

[DOC-826]: https://redpandadata.atlassian.net/browse/DOC-826?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/components/pages/caches/sql.adoc [hunk 1]
```diff
@@ -138,6 +136,8 @@ dsn: foouser:foopassword@tcp(localhost:3306)/foodb
 dsn: postgres://foouser:foopass@localhost:5432/foodb?sslmode=disable
 
 dsn: oracle://foouser:foopass@localhost:1521/service_name
+
+dsn: spanner://projects/fooproject/instances/fooinstance/databases/foodatabase
```
Comment (asimms41) [inline modules/components/pages/caches/sql.adoc:140]: I've added this example. Is it correct. Do you think it's worth adding to the other updated components?

File: modules/components/pages/inputs/sql_select.adoc [hunk 2]
```diff
@@ -108,29 +108,27 @@ input:
...
 
-
 *Type*: `string`
 
-
 Options:
-`mysql`
-, `postgres`
-, `clickhouse`
+`clickhouse`
```
Comment (mihaitodor) [inline modules/components/pages/inputs/sql_select.adoc:114]: Is this omitted comma intentional?

File: modules/components/partials/query_table.adoc [hunk 3]
```diff
@@ -4,27 +4,33 @@
...
+
 | `postgres`
 | Dollar sign (`$`)
 
-| `mssql`
+| `snowflake`
 | Question mark (`?`)
 
-| `sqlite`
+| `snowflake`
```
Comment (mihaitodor) [inline modules/components/partials/query_table.adoc:25]: This seems to be duplicated

**Review by mihaitodor — APPROVED** [review]
LGTM, just a few nits

File: modules/components/partials/query_table.adoc (hunk 3 above)
Comment (Feediver1) [reply modules/components/partials/query_table.adoc:25]: +1

**Review by Feediver1 — APPROVED** [review]
I see a couple updates from eng. Otherwise, lgtm.

File: modules/components/pages/inputs/sql_select.adoc (hunk 2 above)
Comment (asimms41) [reply modules/components/pages/inputs/sql_select.adoc:114]: Hi @mihaitodor, I've looked at the preview and can't see a missing comma: [https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/inputs/sql_select/#driver](https://deploy-preview-122--redpanda-connect.netlify.app/redpanda-connect/components/inputs/sql_select/#driver). Do you agree?

Comment (mihaitodor) [reply modules/components/pages/inputs/sql_select.adoc:114]: Oh, my bad! I didn't realise `clickhouse` ended up being the first item in the list. Thank you for checking! ❤️

**Pull request merged**